### PR TITLE
feat(parser): implement markdown entry extraction and attribute parsing

### LINKED ISSUE #8: Markdown entry extraction
Epic: driftsys/board#14

## Context

Parse Markdown files to identify `- [ID]` entries — list items
starting with `- [TYPE_XYZ_NNNN]` or `- [ID]` (for reference
entries). Extract: ID, title (first line), body (subsequent
paragraphs), trailing `Key: Value` attributes. Preserve source
file and line number for every entry.

Implementation: CommonMark parser (markdown-it or remark) with GFM
table and alert extensions. Only the entry extraction layer is needed
for Sprint — full AST manipulation is PI scope.

## Acceptance Criteria

- [ ] Parse the ADR-001 braking example — three requirements extracted with correct display IDs, titles, bodies, and attribute maps
- [ ] Parse a `references.md` file — reference entries extracted with ID, title, abstract, Document, URL, and optional attributes
- [ ] Source file and line number preserved for every entry

### LINKED ISSUE #9: Attribute parser
Epic: driftsys/board#14

## Context

Parse `Key: Value` trailer blocks at the end of entry bodies.
Recognize authored attributes: `Id`, `Satisfies`, `Derived-from`,
`Labels`, `Document`, `URL`, `Status`, `Superseded-by`. Handle both
backslash-separated and blank-line-separated attribute lines.
Distinguish attributes from body text.

## Acceptance Criteria

- [ ] Attributes with trailing backslash separators parsed correctly
- [ ] Attributes with blank line separators parsed correctly
- [ ] A `Key: Value` line in the middle of body prose is not mistaken for an attribute

### PR DESCRIPTION
**Pull request by stasson**
## What

Implement the Markdown entry extraction parser and attribute block parser — the foundation of the MarkSpec toolchain's parsing pipeline.

## Why

Closes #8
Closes #9

These are the first two parser stories: #8 (Markdown entry extraction) extracts `- [DISPLAY_ID] Title` entry blocks from Markdown files, and #9 (Attribute parser) parses `Key: Value` trailing attribute blocks with backslash separators.

## How

**Attribute parser** (`core/parser/attributes.ts`):
- `parseAttributes(lines)` — parses `Key: Value` lines with optional trailing `\`, recognizes all builtin attribute keys (Id, Satisfies, Derived-from, Labels, Document, URL, Status, Superseded-by)
- `splitBodyAndAttributes(content)` — walks backwards from the end of entry content to find the contiguous trailing attribute block, ensuring `Key: Value` in the middle of body prose is NOT mistaken for an attribute

**Markdown parser** (`core/parser/markdown.ts`):
- Uses unified/remark-parse/remark-gfm to parse CommonMark+GFM into mdast AST
- Walks list items to detect `- [DISPLAY_ID] Title` patterns (handles both text nodes and linkReference nodes from remark)
- Validates display ID format: typed (`TYPE_XYZ_NNNN`) and reference (`[A-Za-z0-9-]+`) patterns
- Extracts body content using source position info, strips list continuation indent
- Delegates attribute extraction to `splitBodyAndAttributes` + `parseAttributes`
- Resolves entry type from typed display IDs, ULID from `Id:` attribute

**Integration**:
- `parser/mod.ts` delegates `parse()` to `parseMarkdown()`
- `compile` CLI command now reads files, parses entries, outputs JSON with `--format json`

**Test coverage** (27 new tests):
- 8 unit tests for attribute parser (backslash separators, whitespace, all known keys, edge cases)
- 13 unit tests for markdown parser (typed entries, reference entries, body extraction, alerts, source location, non-entry items, fixture-based braking requirements)
- 6 E2E acceptance tests (all acceptance criteria from both stories)

## Checklist

- [x] Tests added or updated
- [x] Documentation updated
- [x] All checks pass

### CODE REVIEW

File: packages/markspec/core/parser/markdown.ts [hunk 1]
```diff
@@ -4,3 +4,196 @@
...
+ */
+const TYPED_ID_RE = /^([A-Z]{2,4})_[A-Z]{2,6}_\d{4}$/;
+
+/**
+ * Reference entry display ID pattern: letters, digits, hyphens.
+ */
+const REF_ID_RE = /^[A-Za-z0-9-]+$/;
+
+/**
+ * Match a display ID in `[...]` at the start of a list item paragraph.
```
Comment (stasson) [inline packages/markspec/core/parser/markdown.ts]: **Debt:** `REF_ID_RE` (`[A-Za-z0-9-]+`) matches single characters like `x`, so `- [x] Completed task` with indented body would be incorrectly detected as an entry block. Should exclude task list checkbox patterns (`[x]`, `[ ]`) or require minimum 2-char IDs.

Tracked: will create a debt story.

File: packages/markspec/main.ts [hunk 2]
```diff
@@ -74,7 +74,42 @@ const cli = new Command()
...
+          message: `Failed to read file: ${filePath}: ${err}`,
+          location: undefined,
+        });
+      }
+    }
+
+    const result = { entries: allEntries, diagnostics: allDiagnostics };
+
+    if (_options.format === "json") {
+      console.log(JSON.stringify(result, null, 2));
```
Comment (stasson) [inline packages/markspec/main.ts]: **Debt:** The compile CLI action reads files and calls `parse()` directly instead of delegating to `core/compiler/mod.ts`. This was necessary to unblock E2E tests but breaks the architectural principle that the CLI should delegate to core modules. When the compiler story is implemented, this should call `compiler.compile()` which handles file I/O and validation internally.

Tracked: will create a debt story.

File: packages/markspec/core/parser/markdown.ts [hunk 3]
```diff
@@ -4,3 +4,196 @@
...
+
+  // Get the range from the second child's start to the last child's end
+  const startLine = children[0].position?.start.line;
+  const endLine = children[children.length - 1].position?.end.line;
+
+  if (!startLine || !endLine) return "";
+
+  // Extract the raw lines and strip the list item indentation (typically 2 spaces)
+  const rawLines = lines.slice(startLine - 1, endLine);
+  const stripped = rawLines.map((line) => {
```
Comment (stasson) [inline packages/markspec/core/parser/markdown.ts]: **Debt:** Hardcodes 2-space indent stripping. Works for standard CommonMark list continuation but could be brittle if entries are nested inside deeper list structures, or if editors use different indent widths. Consider detecting actual indent from the list item's position metadata.

Tracked: will create a debt story.

**Review by stasson — COMMENTED** [review]
Thorough review complete. All acceptance criteria for stories #8 and #9 are met. 40/40 tests passing, type-check/lint/format clean.

Three debt items identified with inline comments — creating follow-up stories for each.
